juju/juju
# Verify auth types are supported by Juju when adding cloud.

## Description of change

There are limited number of valid auth types supported by Juju. We should not allowing to load invalid one when adding a new cloud.

## QA steps

1. Have a file cloud config file with an invalid auth-type
2. Run 'juju add-cloud <cloud name> <cloud config from 1>
3. Operation should not succeed. 
Error example: 
ERROR auth types ["user-pass"]  not supported

## Bug reference

Part 2 for https://bugs.launchpad.net/juju/+bug/1590239
